ARCHITECTURE.md

```diff
@@ -0,0 +1,94 @@
+# Architecture
+
+This document describes the high-level architecture of Kedro-Viz. This would be a great starting point for you to get familiar with the codebase and our cross cutting concerns throughout the app. 
+
+See also the contributing docs, which walks through our set of guidelines and recommended best practices for our codebase. 
+
+# High-level Overview
+<!-- Architecture diagram including how standalone app / library entrypoints connect -->
+
+On a high-level, Kedro-viz is a web-app that accepts pipeline data as an input and produces a interactive visualization graph to represent an overview of the current state of the pipeline of the Kedro project. 
+
+More specifically, kedro-viz consumes pipeline data in a specific object format that includes all pipelines, edges an nodes of the Kedro project, meanwhile the Kedro-viz API layer makes request to the Kedro project for further details of a node object, which will be requested and displayed on user clicking a node on the flowchart. (One important thing to note is the concept of a 'node' in Kedro-viz, which is different from the concept of 'node' in Kedro. A 'node' on Kedro-viz referes to an element for display on the flowchart, which could be either a 'database', 'parameter' or 'function' element.)
+
+Kedro-viz is made up of two main parts: the front-end web-app UI that is built on React, and an API layer that pulls data from the Kedro project on which Kedro-viz is running on. 
+
+Currently, the API has 2 endpoints: `/pipeline/<id>` and `/node/<id>`. Each endpoint is called via actions within the app. 
```

Comment:
good point! I agree on the order in which the endpoints should be listed is key. 

Re: the conditions which they are called - I thought about elaborating on that while writing this document, but worry this will go into the territory of too much details, and moreover given that the conditions to call those endpoints could change drastically depending on future work, which would lead to the need to constantly revise this document, hence I was thinking we could keep it as general as possible to only state the avaliable endpoints. What do you think? We could further discuss on that when we pair.

Reply:
I think that the process by which data is requested and loaded into the state is probably the most important thing we could cover in this document. I don't think it's so brittle that it would change in a hurry.

Also fwiw, there are more than 2 endpoints: There's `/main` as well. Also they all begin with `/api`. See the `getUrl` utility.